main.cpp

```diff
@@ -78,14 +128,87 @@ struct List {
     }
 };
 
-void print(List lst) {  // 有什么值得改进的？
-    printf("[");
-    for (auto curr = lst.front(); curr; curr = curr->next.get()) {
-        printf(" %d", curr->value);
+class iterator{
+    private:
+        Node *ptr;
+    public:
+        iterator(Node *ptr = nullptr) : ptr(ptr) {}
+
+        int operator* () const{
+            return ptr->value;
+        }
+
+        Node* operator->(){
+            return ptr;
+        } 
+        
+        iterator& operator++(){
+            ptr = ptr->next.get();
+            return *this;
+        }
+
+        bool operator==(const iterator &other){
+            return ptr==other.ptr;
+        }
+
+        bool operator!=(const iterator &other){
```

Comment:
```suggestion
        bool operator!=(const iterator &other) const {
```
对 other 要 const，对 this 也要 const。